mewarpx/mewarpx/diags_store/checkpoint_diagnostic.py

```diff
@@ -13,32 +15,91 @@
 
 class CheckPointDiagnostic(WarpXDiagnostic):
     def __init__(self, diag_steps,
-                 name=init_restart_util.default_checkpoint_name, **kwargs):
+                 name=init_restart_util.default_checkpoint_name,
+                 clear_old_checkpoints=True, **kwargs):
         """
         This class is a wrapper for creating checkpoints from which
-        to restart a simulation from
+        to restart a simulation from. Adding flux diagnostic data to
+        checkpoints are supported, but since this class has to be initialized
+        before the simulation and flux diagnostics are initialized after
+        the simulation, the user is responsible for adding the ``flux_diag``
+        attribute to this object in a simulation input file.
 
         Arguments:
             diag_steps (int): Run the diagnostic with this period.
                 Also plot on this period if enabled.
             name (str): The name of the diagnostic to be passed into the
                 picmi checkpoint diagnostic.
+            clear_old_checkpoints (bool): If True old checkpoints will be
+                deleted after new ones are created.
             kwargs: For a list of valid keyword arguments see
                 :class:`mewarpx.diags_store.diag_base.WarpXDiagnostic`
         """
-        self.diag_steps = diag_steps
+        self.checkpoint_steps = diag_steps
         self.name = name
+        self.clear_old_checkpoints = clear_old_checkpoints
+        self.flux_diag = None
 
-        super(CheckPointDiagnostic, self).__init__(diag_steps, **kwargs)
+        super(CheckPointDiagnostic, self).__init__(diag_steps+1, **kwargs)
```

Comment:
Is it possible this should be
```suggestion
        if kwargs.pop('diag_step_offset', 1) != 1:
            raise ValueError(
                "diag_step_offset in CheckPointDiagnostic must be 1.")
        super(CheckPointDiagnostic, self).__init__(
            diag_steps=diag_steps, diag_step_offset=1, **kwargs)
```
Otherwise it seems like this will run 1 step after the checkpoint file, then 2 steps  after the checkpoint file is written, then 3 after etc.

Reply:
Oh yeah, good point!